DocPrepare/PrepareDocs.py

```diff
@@ -0,0 +1,137 @@
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+import os
+import json
+import yaml
+
+
+class PrepareDocs:
+
+    def __init__(self, agency_directory, unique_parser=None,
+                 s3_bucket_name=None):
+        """
+        agency_directory: directory of a specific office or agency
+        unique_parser: optional parser function for document metadata
+        not extracted with Tika
+        """
+        self.agency_directory = agency_directory
+        self.unique_parser = unique_parser
+
+        if s3_bucket_name:
+            self.s3_bucket = S3Connection().get_bucket(s3_bucket_name)
+        else:
+            self.s3_bucket = None
+
+    def parse_date(self, date):
+        """ Given string like this: 2013-03-20T17:11:17Z returns date in
+        format YYYY-MM-DD"""
+
+        if date:
+            return date.split('T')[0]
+
+    def parse_tika_metadata(self, metadata_file, metadata):
+        """ Parses metadata from created by Tika """
+
+        with open(os.path.join(metadata_file), 'r') as f:
+            try:
+                tika_metadata = json.loads(f.read())
+            except:
+                tika_metadata = {}
+
+            if not metadata.get('file_type'):
+                metadata['file_type'] = tika_metadata.get(
+                    ['dc:format'], '').replace('application/', '')
+            if not metadata.get('date_released'):
+                metadata['date_released'] = self.parse_date(
+                    tika_metadata.get('Last-Save-Date'))
+            if not metadata.get('title'):
+                metadata['title'] = tika_metadata.get('title')
+
+            metadata['pages'] = tika_metadata.get('xmpTPg:NPages')
+            metadata['date_created'] = self.parse_date(
+                tika_metadata.get('meta:creation-date'))
+
+            return metadata
+
+    def prepare_metadata(self, root, base_file):
+        """ Prepares metadata from Tika metadata file, the file location,
+        and unique parser if available """
+
+        if self.unique_parser:
+            metadata = self.unique_parser(
+                os.path.join(root, base_file + ".json"))
+        else:
+            metadata = {}
+
+        metadata = self.parse_tika_metadata(
+            os.path.join(root, base_file + "_metadata.json"),
+            metadata=metadata)
+
+        return metadata
+
+    def prepare_file_location(self, metadata, root, base_file):
+        """ Adds file location to metadata """
+
+        if self.s3_bucket:
+
+            # Upload pdf
```

Comment:
I don't think this function should care about the filetype. Just upload whatever file is provided.